core/src/com/unciv/logic/civilization/TechManager.kt

```diff
@@ -210,7 +210,20 @@ class TechManager {
         overflowScience = limitOverflowScience(overflowscience)
         addTechnology(currentTechnology)
     }
+    fun getExtraScience(scienceGet : Int) {
```

Comment:
A lot of this code is copy-pasted from nextTurn, we should use this function when calling nextTurn to avoid code duplications

Reply:
`fun getExtraScience`is actually copy-pasted from fun hurryResearch() as you can see in the comment.

you mean I should edit this code to get extra science when ending the turn? 

in the original game, the bonus science point is get when right after the scientific buildings/wonders are finished.

Reply:
Ye - if the same code is used to add science on hurry research and on next turn, they should use a common function like addScience(amount) or something

Reply:
Okay I got It. I will merge `hurryResearch()` and `addScience(amount)`.

Reply:
Done with merging two function into one function. `addScience(amout:Int)`